AK/URLParser.cpp

```diff
@@ -50,17 +50,316 @@ static Optional<DeprecatedString> parse_ipv4_address(StringView input)
     return input;
 }
 
+// https://url.spec.whatwg.org/#concept-ipv6-serializer
+static void serialize_ipv6_address(Array<u16, 8> const& address, StringBuilder& output)
```

Comment:
Why are you taking a `StringBuilder`?
Making the function return a `String` will make it both easier for the caller and closer to spec.

Reply:
The point of taking a `StringBuilder` was to eventually call this from `URL::serialize()` which uses a `StringBuilder` but yeah, we can just change it to a `StringBuilder` when we actually do that. :)